src/modules/dna-service/dna.background.ts

```diff
@@ -248,7 +253,49 @@ export class DnaService {
     return browserStorage.set(ACTION_QUEUE_KEY, []);
   }
 
+  async registerAllWallets() {
+    const isInvoked = await browserStorage.get<boolean>(
+      REGISTER_ALL_WALLETS_INVOKED_KEY
+    );
+    if (isInvoked) {
+      return;
+    }
+
+    const wallet = this.getWallet();
+    const walletGroups = await wallet.uiGetWalletGroups({
+      context: INTERNAL_SYMBOL_CONTEXT,
+    });
+
+    const ownedAddresses =
+      walletGroups
+        ?.filter((group) => !isReadonlyContainer(group.walletContainer))
+        ?.flatMap((group) =>
+          group.walletContainer.wallets.map((wallet) => ({
+            address: wallet.address,
+            origin: group.origin || undefined,
+          }))
+        ) || [];
+
+    let didFail = false;
+    for (const { address, origin } of ownedAddresses) {
+      try {
+        await this.registerWallet({
+          params: { address, origin },
+        });
+      } catch (error) {
+        didFail = true;
+        console.warn('registerWallet error', error); // eslint-disable-line no-console
+      }
+    }
```

Comment:
Do these calls need to be one after the other? Can we use `Promise.all` instead to make them in parallel?

Reply:
Yep, you're right. If backend fails to register a single wallet it doesn't mean we shouldn't attempt to register others. I'll update it to `Promise.all`

Reply:
I increased number of retries to 3, added `POST` to the list "retriable" methods. Now `registerWallet` requests are sent in parallel. Added clarifying comment as well ✅  Please check it out